app/src/main/java/com/example/ihonboarding/repository/auth/AuthRepositoryImpl.kt

```diff
@@ -0,0 +1,127 @@
+                    response.body()?.let { authTokenDto ->
+                        val result = saveAuthToken(authTokenDto, authLoginRequest.email)
+                        if (result < 0) {
+                            Timber.e("Couldn't save an auth token into db.")
+                        }
+                        saveAuthenticatedUserToDataStorePrefs(authLoginRequest.email)
+                        authTokenDto.let {
+                            Resources.success(
+                                authTokenDtoMapper.mapToDomainModel(it)
+                            )
+                        }
+                    } ?: returnUnknownError()
+                } else {
+                    response.errorBody()?.let { responseBody ->
+                        val errorMessage =
+                            JSONObject(responseBody.charStream().readText()).getString("error")
+                        Resources.error(
+                            errorMessage,
+                            AuthToken(
+                                errorResponse = StateResponse(
+                                    message = errorMessage,
+                                    errorResponseType = ResponseType.Dialog
+                                )
+                            )
+                        )
+                    } ?: returnUnknownError()
+                }
+            }
+
+    }
+
+    override suspend fun checkPrevAuthUser(): Flow<Resources<AuthToken>> {
+        return authPrefsManager.preferencesFlow
+            .map { email ->
+                if (email.isBlank()) {
+                    Timber.d("checkPreviousAuthUser: No previously authenticated user found.")
+                    returnNoAuthTokenFound()
+                }
+
+                email.let {
+                    authTokenDao.searchByEmail(email)?.let { authTokenEntity ->
+                        authTokenEntity.id?.let {
+                            if (it > -1) {
+                                authTokenDao.searchById(it)?.let { authToken ->
+                                    if (authToken.token != null) {
+                                        Resources.success(
+                                            authTokenEntityMapper.mapToDomainModel(authToken)
+                                        )
+                                    } else returnNoAuthTokenFound()
+                                }
+                            } else returnNoAuthTokenFound()
+                        }
+                    }
+                } ?: returnNoAuthTokenFound()
+
+            }
+
+    }
+
+    private suspend fun saveAuthenticatedUserToDataStorePrefs(email: String) {
+        authPrefsManager.saveAuthenticatedUserToDataStore(email)
+        Timber.d("Email in ds: ${authPrefsManager.readPreviousAuthUserEmail()}")
+    }
+
+    private suspend fun saveAuthToken(authTokenDto: AuthTokenDto, email: String): Long =
+        authTokenDao.insert(
```

Comment:
I don't see any logic for token encryption here. Is that will included in this PR or another?